src/convertToObject.js

```diff
@@ -5,8 +5,21 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((block) => block.trim())
+    .filter((block) => block.includes(':'))
+    .reduce((cssProperties, block) => {
+      const [key, value] = block
+        .split(':')
+        .map((splitBlock) => splitBlock.trim());
+
+      cssProperties[key] = value;
+
+      return cssProperties;
+    }, {});
 }
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. While the code correctly uses iteration methods such as `split`, `map`, `filter`, and `reduce`, there is an issue with the `reduce` callback function. The current implementation is mutating the `cssProperties` object in each iteration, which is not an issue in itself, but could be improved by using a more functional approach where you don't manually mutate objects.